### PR TITLE
qdwh

### LINKED ISSUE #13731: qdwh

### PR DESCRIPTION
**Pull request by Burhankhalid**
close #13731

### CODE REVIEW

**Comment by Burhankhalid** [comment]
Hey, @Ookamice I've created this new Pull Request for further modifications. Kindly guide me on how to further improve my code to get this PR merged.

Thanks again for your assistance.

**Comment by Ookamice** [comment]
@Burhankhalid 
Just had a look in the PR tests, looks like all the tests are failing because ` hypothesis.errors.InvalidArgument: Cannot apply @given to a function with defaults.` I think it would be great if you could look at how the other jax.lax.linalg tests are done as your test logic seems overly complicated. 

Note that the test is meant to compare the ivy frontend you've written to the existing jax.lax.linalg.qdwh so I'm not fully sure why you are doing so much computation..

Do correct me if I've missed anything and let me know if you got any more questions! 😃

**Comment by Burhankhalid** [comment]
Hi, it seems that @KareemMAX has adopted these changes because the function is showing him as the collaborator on these functions now.

Screenshot (93)
Screenshot (94)

**Comment by KareemMAX** [comment]
Hi @Burhankhalid, I'm frequently mentioned on code because I worked on some linting procedures, but I didn't write that function 😅. You might need to go back with the blame history a little bit

**Comment by KareemMAX** [comment]
@Burhankhalid I tried to trace the function you mentioned me in. I guess I didn't make any changes in at as it's actually on your branch https://github.com/Burhankhalid/ivy/blame/linalg/ivy/functional/frontends/jax/lax/linalg.py#L42

**Comment by Burhankhalid** [comment]
Hi, @Ookamice can you review this now and let me know any further improvements? Thanks.

**Comment by Ookamice** [comment]
@Burhankhalid 
Hiya, glad to see you're still working on it. I've read through and would like to propose some changes that I'll list below:
1) Please do not use np_frontend and use the ivy implementations instead
2) Could you look into how others used `helpers.test_frontend_function` to obtain `ret, frontend_ret` which greatly simplifies test logic
3) Along with 2), it may be a good idea to look into atol and rtol in your decorators due to the nature of this function. (and I can see you've already used them in your assert logic 😃)
4) Also in your current testing logic, there is an error about unexpected keyword `dtype_and_x`

Lastly, feel free to ask any other questions that you may have 👍

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
